x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/expandable_section/expandable_section_results.tsx

```diff
@@ -127,6 +186,104 @@ export const ExpandableSectionResults: FC<ExpandableSectionResultsProps> = ({
   );
   const analysisType =
     jobConfig && jobConfig.analysis ? getAnalysisType(jobConfig.analysis) : undefined;
+
+  const generateDiscoverUrl = useCallback(
+    async (rowIndex: number) => {
+      const item = tableItems[rowIndex];
+
+      if (discoverLocator !== undefined) {
+        const url = await discoverLocator.getRedirectUrl({
+          indexPatternId: dataViewId,
+          timeRange: data.query.timefilter.timefilter.getTime(),
+          filters: data.query.filterManager.getFilters(),
+          query: {
+            language: SEARCH_QUERY_LANGUAGE.KUERY,
+            // Filter for all visible column values of supported types - except the results field values
+            query: indexData.visibleColumns
+              .filter(
+                (column) =>
+                  item[column] !== undefined &&
+                  (typeof item[column] === 'string' || typeof item[column] === 'number') &&
+                  !column.includes(resultsField!)
+              )
+              .map((column) => `${escapeKuery(column)}:${escapeKuery(String(item[column]))}`)
+              .join(' and '),
+          },
+        });
+
+        return url;
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [indexData?.visibleColumns, discoverLocator, dataViewId, resultsField, tableItems]
+  );
+
+  const trailingControlColumns: EuiDataGridProps['trailingControlColumns'] = [
+    {
+      id: 'actions',
+      width: 60,
+      headerCellRender: () => (
+        <FormattedMessage
+          id="xpack.ml.dataframe.analytics.exploration.dataGridActions.columnTitle"
+          defaultMessage="Actions"
+        />
+      ),
+      rowCellRender: function RowCellRender({ rowIndex }) {
+        const [isPopoverVisible, setIsPopoverVisible] = useState(false);
+        const closePopover = () => setIsPopoverVisible(false);
+
+        const actions = [
+          <EuiContextMenuItem
+            icon="discoverApp"
+            key="custom-discover-url"
+            disabled={discoverUrlError !== undefined}
+            onClick={async () => {
+              const openInDiscoverUrl = await generateDiscoverUrl(rowIndex);
+              if (openInDiscoverUrl) {
+                application.navigateToUrl(openInDiscoverUrl);
+              }
+            }}
+          >
+            {discoverUrlError ? (
+              <EuiToolTip content={discoverUrlError}>
+                <FormattedMessage
+                  id="xpack.ml.dataframe.analytics.exploration.dataGridActions.viewInDiscover"
+                  defaultMessage="View in Discover"
+                />
+              </EuiToolTip>
+            ) : (
+              <FormattedMessage
+                id="xpack.ml.dataframe.analytics.exploration.dataGridActions.viewInDiscover"
+                defaultMessage="View in Discover"
+              />
+            )}
+          </EuiContextMenuItem>,
+        ];
+
+        return (
+          <>
```

Comment:
looks like this fragment isn't needed

Reply:
Removed in 1664d294c98a5c1b8c9e86cfc426be49b467bd40